src/engine/commands/generate/commands/trigger.ts

```diff
@@ -0,0 +1,65 @@
+import * as yargs from "yargs";
+
+import { Context } from "../../../../common/context";
+import { translations } from "../../../../common/translations";
+import { ExtensionType, SyntaxType, TriggerOperation, TriggerType } from "../../../../interfaces/Extensions";
+import { ProjectController } from "../../../controllers/projectController";
+
+type TiggerParams = {
+  name: string,
+  type?: string,
+  operation?: string,
+  mocks: boolean,
+  syntax: SyntaxType,
+};
+
+export default {
+  command: "trigger <name>",
+
+  handler: async (params: TiggerParams, context: Context) => {
+    let { name, type, operation, mocks, syntax } = params;
+
+    if (operation && !/[\w\d]+\.(create|update|delete)/.test(operation)) {
+      throw new Error(translations.i18n.t("generate_trigger_invalid_operation"));
+    }
+
+    ProjectController.generateFunction(context, {
+      type: ExtensionType.trigger,
+      name,
+      mocks,
+      syntax,
+    }, {
+      type,
+      operation,
+    });
+  },
+
+  describe: translations.i18n.t("generate_trigger_describe"),
+
+  builder: (args: yargs.Argv): yargs.Argv => args
+    .usage(translations.i18n.t("generate_trigger_usage"))
+    .option("type", {
+      alias: "t",
+      describe: translations.i18n.t("generate_trigger_type_describe"),
+      type: "string",
+      choices: Object.values(TriggerType),
+    })
+    .option("operation", {
+      alias: "o",
+      describe: translations.i18n.t("generate_trigger_operation_describe"),
+      type: "string",
```

Comment:
@zouxuoz Why is there no choices for operation like for type?

Reply:
@vorobeez because it could be `AnyTableName.(create|update|delete)`

Reply:
I see, ok